mlflow/data/dataset.py

```diff
@@ -40,18 +40,18 @@ def _compute_digest(self) -> str:
         """
 
     @abstractmethod
-    def _to_dict(self, base_dict: Dict[str, str]) -> Dict[str, str]:
-        """
-        Args:
-            base_dict: A string dictionary of base information about the
-                dataset, including: name, digest, source, and source
-                type.
+    def to_dict(self) -> Dict[str, str]:
```

Comment:
classes that are overridden in subclasses should be public.

Reply:
(1) makes sense to me. i dont feel strongly about (2), but it seems like a resonable change